src/test/java/net/smartcosmos/userdetails/resource/AuthenticateResourceTest.java

```diff
@@ -0,0 +1,286 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.mock.http.MockHttpOutputMessage;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import net.smartcosmos.test.config.UserDetailsResourceTestConfiguration;
+import net.smartcosmos.userdetails.domain.UserDetails;
+import net.smartcosmos.userdetails.domain.rest.AuthenticateDetails;
+import net.smartcosmos.userdetails.domain.rest.AuthenticateRequest;
+import net.smartcosmos.userdetails.service.AuthenticateUserService;
+
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.isEmptyOrNullString;
+import static org.hamcrest.Matchers.not;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.when;
+import static org.springframework.http.MediaType.APPLICATION_JSON_UTF8;
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import static net.smartcosmos.domain.rest.ErrorResponse.CODE_ERROR;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = { UserDetailsResourceTestConfiguration.class })
+@WebAppConfiguration
+@ActiveProfiles("test")
+public class AuthenticateResourceTest {
+
+    @Autowired
+    AuthenticateUserService authenticateUserService;
+
+    // region setup
+
+    @Autowired
+    protected WebApplicationContext webApplicationContext;
+    protected MockMvc mockMvc;
+    protected HttpMessageConverter mappingJackson2HttpMessageConverter;
+
+    @Autowired
+    void setConverters(HttpMessageConverter<?>[] converters) {
+
+        this.mappingJackson2HttpMessageConverter = Arrays
+            .stream(converters)
+            .filter(hmc -> hmc instanceof MappingJackson2HttpMessageConverter)
+            .findAny()
+            .get();
+
+        Assert.assertNotNull("the JSON message converter must not be null",
```

Comment:
That is nice to ensure but if someone is messing around with this we have big problems.